clients/ui/frontend/src/app/hooks/useNotification.ts

```diff
@@ -0,0 +1,109 @@
+import React, { useContext } from 'react';
+import { AlertVariant } from '@patternfly/react-core';
+import { NotificationActionTypes, NotificationContext } from '~/app/context/NotificationContext';
+
+enum NotificationTypes {
+  SUCCESS = 'success',
+  ERROR = 'error',
+  INFO = 'info',
+  WARNING = 'warning',
+}
+
+type NotificationProps = (title: string, message?: React.ReactNode) => void;
+
+type NotificationRemoveProps = (id: number | undefined) => void;
+
+type NotificationTypeFunc = {
+  [key in NotificationTypes]: NotificationProps;
+};
+
+interface NotificationFunc extends NotificationTypeFunc {
+  remove: NotificationRemoveProps;
+}
+
+export const useNotification = (): NotificationFunc => {
+  const { dispatch } = useContext(NotificationContext);
+  // need to move this count somewhere else since it will reset on every new useNotification instance (like switching pages)
+  let notificationCount = 0;
```

Comment:
Why don't you move the notification count to the context itself?

Reply:
Why don't I......